src/components/widgets/NumberWidget.js

```diff
@@ -106,7 +106,9 @@ export default class NumberWidget extends React.Component {
   }
 
   onChange = e => {
-    this.setState({value: e.target.value, displayValue: e.target.value});
+    const value = e.target.value;
+    this.setState({value: value, displayValue: value});
+    this.setValue(value);
```

Comment:
If we set the value here it will set the format to `.toFixed` so it won't let you keep typing. If we just let the user write and it will just take the actual value onBlur (or onKeyPress enter) and just leave the onChange to change the display value.